### PR TITLE
retry logic galore #370

### PR DESCRIPTION
**Pull request by theotherian**

### CODE REVIEW

File: stream-parent/stream-client/src/main/java/org/interledger/stream/sender/DefaultBackoffController.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+      return response;
+    }
+
+    public int rejectsReceived() {
+      return rejects.get();
+    }
+  }
+
+  @Override
+  public InterledgerResponsePacket sendWithBackoff(final Link link, final InterledgerPreparePacket preparePacket,
```
Comment (nhartner) [inline stream-parent/stream-client/src/main/java/org/interledger/stream/sender/DefaultBackoffController.java]: I think that the preparePacket.expiresAt would need to be updated/extended before every retry attempt. When SimpleStreamSender creates the preparePacket it sets the expiresAt to 30s in the future. If you configured back such that it retries after 1s, 2s, 4s, 8s, 16s then the logic attempts retry a preparePacket that is already expired.

Comment (theotherian) [reply stream-parent/stream-client/src/main/java/org/interledger/stream/sender/DefaultBackoffController.java]: Good catch; totally spaced that behavior. Thanks!

File: stream-parent/stream-client/src/main/java/org/interledger/stream/sender/DefaultBackoffController.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+import io.github.resilience4j.retry.Retry;
+import io.github.resilience4j.retry.RetryConfig;
+
+import java.time.Duration;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Function;
+
+public class DefaultBackoffController implements BackoffController {
```
Comment (sappenin) [inline stream-parent/stream-client/src/main/java/org/interledger/stream/sender/DefaultBackoffController.java]: If we keep this PR, let's add some Javadoc to outline what the default backoff behavior is?

File: stream-parent/stream-client/src/main/java/org/interledger/stream/sender/DefaultBackoffController.java (hunk 1 above)
Comment (sappenin) [reply stream-parent/stream-client/src/main/java/org/interledger/stream/sender/DefaultBackoffController.java]: I was expecting that if a packet fails, we would consider it a reject, and the "retry" would just be a new packet. Is the design here to do something different?

**Pull request closed without merging**